internal/clients/clients.go

```diff
@@ -0,0 +1,108 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/clientcmd"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	clusterv1beta1 "open-cluster-management.io/api/cluster/v1beta1"
+	clusterv1beta2 "open-cluster-management.io/api/cluster/v1beta2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+)
+
+const (
+	ListRequestTimeout   = 30 * time.Second
+	SingleRequestTimeout = 10 * time.Second
+
+	secretTypeLabel      = "hive.openshift.io/secret-type"
+	secretTypeLabelValue = "kubeconfig"
+)
+
+// NewClientForHub creates a new client for the hub cluster
+func NewClientForHub() (client.Client, error) {
+	conf, err := config.GetConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get config: %w", err)
+	}
+
+	c, err := client.New(conf, client.Options{Scheme: GetSchemeForHub()})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client: %w", err)
+	}
+
+	return c, nil
+}
+
+// GetSchemeForHub returns the scheme for the hub cluster client
+func GetSchemeForHub() *runtime.Scheme {
+	scheme := runtime.NewScheme()
+	utilruntime.Must(corev1.AddToScheme(scheme))
+	utilruntime.Must(clusterv1.AddToScheme(scheme))
+	utilruntime.Must(clusterv1beta1.AddToScheme(scheme))
+	utilruntime.Must(clusterv1beta2.AddToScheme(scheme))
```

Comment:
Something to note that we need to apply the RBACs to fetch the data (managedcluster stuff maybe not since we get admin kubeconfig)...depending on the service account. I think we can all share the same account that operators are using but something to think about.

Reply:
Since we are not ready to integrate/deploy to cluster yet it's not really important but maybe a todo to note this.

Reply:
but not here, we could create a story and write that comment